backend/src/data/repositories/slc-function/slc-function.repository.ts

```diff
@@ -0,0 +1,82 @@
+import { SLCFunction as SLCFunctionM } from '~/data/models/models';
+import { FunctionTableField } from '~/data/models/slc-function/slc-function-table-field.enum';
+import { SLCFunction as SLCFunctionEntity } from '~/services/slc-function/slc-function.entity';
+
+type Constructor = {
+  SLCFunctionModel: typeof SLCFunctionM;
+};
+
+class SLCFunction {
+  #SLCFunctionModel: typeof SLCFunctionM;
+
+  constructor({ SLCFunctionModel }: Constructor) {
+    this.#SLCFunctionModel = SLCFunctionModel;
+  }
+
+  async getByName({
+    workersIds,
+    name,
+  }: {
+    workersIds: string[];
+    name: string;
+  }): Promise<SLCFunctionEntity | null> {
```

Comment:
I fear no man. But when I see function that accepts an **array** of workers and return **one** element, I'm really scared. Such methods can be very hard to debug